File: src/Implicit/ImplicitGain.cs
using System;

namespace TinkerWorX.AccidentalNoiseLibrary
{
    public sealed class ImplicitGain : ImplicitModuleBase
    {
        public ImplicitGain(ImplicitModuleBase source, double gain = 0.00)
        {
            this.Source = source;
            this.Gain = new ImplicitConstant(gain);
        }

        public ImplicitGain(ImplicitModuleBase source, ImplicitModuleBase gain)
        {
            this.Source = source;
            this.Gain = gain;
        }

        public ImplicitModuleBase Source { get; set; }

        public ImplicitModuleBase Gain { get; set; }

        public override double Get(double x, double y)
        {
            return Utilities.Gain(this.Gain.Get(x, y), this.Source.Get(x, y));
        }

        public override double Get(double x, double y, double z)
        {
            return Utilities.Gain(this.Gain.Get(x, y, z), this.Source.Get(x, y, z));
        }

        public override double Get(double x, double y, double z, double w)
        {
            return Utilities.Gain(this.Gain.Get(x, y, z, w), this.Source.Get(x, y, z, w));
        }

        public override double Get(double x, double y, double z, double w, double u, double v)
        {
            return Utilities.Gain(this.Gain.Get(x, y, z, w, u, v), this.Source.Get(x, y, z, w, u, v));
        }
    }
}

File: src/implicit/ImplicitGain.java
package implicit;

import util.Maths;

public final class ImplicitGain extends ImplicitModuleBase {
    public ImplicitGain(final ImplicitModuleBase source, final double gain) {
        this.Source = source;
        this.Gain = new ImplicitConstant(gain);
    }

    public ImplicitGain(final ImplicitModuleBase source, final ImplicitModuleBase gain) {
        this.Source = source;
        this.Gain = gain;
    }

    private ImplicitModuleBase Source;

    private ImplicitModuleBase Gain;

    public ImplicitModuleBase getSource() {
        return this.Source;
    }

    public void setSource(final ImplicitModuleBase source) {
        this.Source = source;
    }

    public ImplicitModuleBase getGain() {
        return this.Gain;
    }

    public void setGain(final ImplicitModuleBase gain) {
        this.Gain = gain;
    }

    @Override
    public double get(final double x, final double y) {
        return Maths.Gain(this.Gain.get(x, y), this.Source.get(x, y));
    }

    @Override
    public double get(final double x, final double y, final double z) {
        return Maths.Gain(this.Gain.get(x, y, z), this.Source.get(x, y, z));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w) {
        return Maths.Gain(this.Gain.get(x, y, z, w), this.Source.get(x, y, z, w));
    }

    @Override
    public double get(final double x, final double y, final double z, final double w, final double u, final double v) {
        return Maths.Gain(this.Gain.get(x, y, z, w, u, v), this.Source.get(x, y, z, w, u, v));
    }
}
